### DISCUSSION TITLE
Allocate maximum height for scroll area

### DISCUSSION THREAD

**Discussion by feelingnothing** [emilk/egui#939] [Q&A]
Code:
```rust
egui::CentralPanel::default().show(ctx, |ui| {
    ui.horizontal(|ui| {
        ui.vertical(|ui| {
            ui.strong("Settings:");
            if ui.button("Add Line").clicked() {
                self.log.push(format!("{}", self.log.len() + 1));
            }
        });
        ui.vertical(|ui| {
            ui.strong("Log:");
            Frame::dark_canvas(ui.style()).show(ui, |ui| {
                egui::ScrollArea::vertical().stick_to_bottom().auto_shrink([false; 2]).show_rows(
                    ui, ui.fonts()[TextStyle::Body].row_height(), self.log.len(),
                    |ui, range| {
                        for i in range { ui.label(i.to_string()); };
                    },
                );
            });
        });
    });
});
```
Outputs:
![image](https://user-images.githubusercontent.com/47575622/145111770-0f123135-7d21-42fc-b0e5-d480d71bf6e4.png)

How can I stretch log area to a available height? Tried allocating ui with available size for `dark_canvas`, tried allocation space with `available_size` in various scopes (after canvas, in canvas after scroll area, after `vertical`) and all of it didn't work, I just don't undestand how layouts in reactive gui frameworks work

**Comment by NickNick** [comment]
I found the answer on https://github.com/emilk/egui/discussions/1381. However, I ended up replacing`ui.horizontal` with `ui.horizontal_top`.
